app/common/elasticsearchclient.py

```diff
@@ -243,17 +243,27 @@ def best_hit_search(self,
         #there are len(params.q) responses - one per query
         for i in range(len(results['responses'])):
             name = params.q[i]  #even though we are guaranteed that responses come back in order, and can match query to the result - this might be convenient to have       
+            print "name =" + name 
             lower_name = name.lower()
             res = results['responses'][i]
             exact_match = False
             if(res['hits']['total']>0):
                 hit = res['hits']['hits'][0] #expect either 1 result or none
                 highlight = ''
+                fields = kwargs['fields'];
                 if 'highlight' in hit:
                     highlight = hit['highlight']
-                if(lower_name == hit['_source']['approved_symbol'].lower() or lower_name == hit['_id'].lower):
+                
+                id_lower = hit['_id'].lower()
+                type_ = hit['_type']
+                if (lower_name == id_lower):
                     exact_match = True
-                datapoint = dict(type=hit['_type'],
+                elif (type_ == 'search-object-target' and lower_name == hit['_source']['approved_symbol'].lower()):
```

Comment:
It would be better to store the type into an enum at the beginning of the file (like the `FreeTextFilterOptions` class above) and check for a parameter of that class. In this way if we change those strings we have a single place to replace them